File: src/main/java/com/fractalwrench/bpp/BppDialect.java
package com.fractalwrench.bpp;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

final class BppDialect {

    private final Map<String, String> map;
    private final String dialectName;

    BppDialect(Map<String, String> map, String dialectName) {
        this.map = new HashMap<>(map);
        this.dialectName = dialectName;
    }

    Map<String, String> getMap() {
        return Collections.unmodifiableMap(map);
    }

    String getDialectName() {
        return dialectName;
    }

    @Override
    public String toString() {
        return "BppDialect{" +
                "dialectName='" + dialectName + '\'' +
                ", map=" + map +
                '}';
    }
}

File: src/main/java/com/fractalwrench/bpp/BppDialect.kt
package com.fractalwrench.bpp

import java.util.Collections
import java.util.HashMap

internal class BppDialect(map: Map<String, String>, val dialectName: String) {

    private val map: Map<String, String>

    init {
        this.map = HashMap(map)
    }

    fun getMap(): Map<String, String> {
        return Collections.unmodifiableMap(map)
    }
}
